pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -197,3 +199,33 @@ func hasDeploymentTimedOut(deployment *appsv1.Deployment) bool {
 	}
 	return false
 }
+
+func (c *Reconciler) reconcileSecret(ctx context.Context, rev *v1.Revision) error {
+	ns := rev.Namespace
+	secretName := ns + "-" + networking.ServingCertName
+	logger := logging.FromContext(ctx)
+	logger.Info("Reconciling Secret: ", secretName)
+
+	secret, err := c.kubeclient.CoreV1().Secrets(ns).Get(ctx, secretName, metav1.GetOptions{})
+	if apierrs.IsNotFound(err) {
+		namespace, err := c.kubeclient.CoreV1().Namespaces().Get(ctx, ns, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get Namespace %q: %w", secretName, err)
+		}
+		secret, err = c.createSecret(ctx, namespace)
```

Comment:
```suggestion
		secret, err = c.createSecret(ctx, namespace)
```
```suggestion
				if secret, err = c.createSecret(ctx, namespace); err != nil {
			return fmt.Errorf("failed to create Secret %q: %w", secretName, err)
		}
```

Reply:
No, we cannot use the style here as the `secret` is used later around L225.

Reply:
`secret` is already declared before `if apierrs.IsNotFound(err) {` and value is re-assigned in that if statement no?

Reply:
Ah, yes. it is possible to write the one line code. updated.